### ISSUE TITLE
Package for pfSense to utilize Tailscale

### ISSUE THREAD

**Issue by mspencerl87** [tailscale/tailscale#787]
Was asked to open this Bug report by BradFitz
https://www.reddit.com/r/Tailscale/comments/iukyaf/network_bridging_pfsenseopnsense/ 

**Describe the bug**
Create Package for use on OPNsense and PFsense systems.
Also to allow bridging connections within the application

**To Reproduce**
No errors

**Expected behavior**
Currently a similar software called Zerotier has a package for OPNsense that allows the router to be a member of the network.
Also advanced configuration exists to allow bridging the LAN network with the Zerotier network.
Would like a similar solution with Tailscale.

**Version information:**
Latest builds for both systems

**Additional context**
No additional context

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_2i2j5)

**Comment by newmy-de** [comment]
installing tailscale on opnsense 20.7.4 is possible in this way:

login via ssh.
```
# opnsense-code ports
# cd /usr/ports/security/tailscale
# make install
# service tailscaled enable
# service tailscaled start
# tailscale up
```

copy the url and login with your credential.
your opnsense is now connected with your account.

unfortunately a connection via the tailscale ip is (for me) not possible, so something is missing, maybe to the old 1.0.3. version of tailscale in the freebsd ports.

**Comment by Strider3000** [comment]
tailscale 1.2.0 for freebsd breaks at the make install step:

```
# cmd/link/internal/ld
src/cmd/link/internal/ld/symtab.go:333:16: internal compiler error: '(*Link).symtab': value v5085 still has 65536 uses

Please file a bug report including a short program that triggers the error.
https://golang.org/issue/new

go tool dist: FAILED: /usr/obj/usr/ports/lang/go/work/go/pkg/tool/freebsd_amd64/go_bootstrap install -gcflags=all= -ldflags=all= -v -i cmd/asm cmd/cgo cmd/compile cmd/link: exit status 2
*** Error code 2

Stop.
make[1]: stopped in /usr/ports/lang/go
*** Error code 1

Stop.
make: stopped in /usr/ports/security/tailscale
```
So... can we get an official opnsense package?

**Comment by mspencerl87** [comment]
Thanks @newmy-de does this show up in the GUI after making?

**Comment by newmy-de** [comment]
> Thanks @newmy-de does this show up in the GUI after making?

yes. my opnsense showd up in the tailscale administration and partly in opnsense GUI.
the tailscale networkport shows up e.g. Interface Assignments (but i was not able to rally assin an interface to it.),
in system - routes - status all my systems are shown with gateway tailscale. 

I can ping outside and also doing ssh and so. but getting access from another TS client to the opnsense was not functional.

Every access via the 100.x.x.x was blocked by the DefaultDenyRule.

I tried several - trying to make the tailscale interface schowing up in opnsense GUI but every time the whole configuration gets corruptet. 
So i needed an other way ...

Fixing that requests a firwall rule inside the Floating Section passing the traffic "inside" the tailscale network.
I used an alias for definig the tailscale network:

Alias:
**Name | Type | Description | Content**
NET_tailscale | Network(s) | tailscale Network | 100.0.0.0/8

Rule in the Floating Section:
**Protocol | Source | Port | Destination | Port | Gateway | Schedule | Description**
IPv4 TCP/UDP | NET_tailscale | * | This Firewall | * | Null4 | * | tailscale

With this setting i'm able to get access from my phone via LTE to the webportal of the opnsense inside a NATed network

LTE =german telekom
VDLS = german telekom --> Draytek 130 Modem --> UDM Pro --> opnsense
opnsense = OPNsense 20.7.4-amd64 | FreeBSD 12.1-RELEASE-p10-HBSD | OpenSSL 1.1.1h 22 Sep 2020 | on old sophos SG210 
tailscale = 1.0.3 installed like i described it before

**Comment by newmy-de** [comment]
updated my 1.0.3 to 1.2.0 without errors seen by @Strider3000 using:

**root@OPNsense**:/usr/ports/security/tailscale # make install clean
===>  Cleaning for go-1.15.4,1
===>  Cleaning for tailscale-1.2.0
**root@OPNsense**:/usr/ports/security/tailscale # make install clean

+++ many many lines +++

endet in:
===>   An older version of tailscale is already installed (tailscale-1.0.3)
      You may wish to ``make deinstall'' and install this port again
      by ``make reinstall'' to upgrade it properly.
      If you really wish to overwrite the old port of tailscale
      without deleting it first, set the variable "FORCE_PKG_REGISTER"
      in your environment or the "make install" command line.
*** Error code 1


Stop.
make: stopped in /usr/ports/security/tailscale

**root@OPNsense**:/usr/ports/security/tailscale # make deinstall
===>  Deinstalling for tailscale
===>   Deinstalling tailscale-1.0.3
Updating database digests format: 100%
Checking integrity... done (0 conflicting)
Deinstallation has been requested for the following 1 packages (of 0 packages in the universe):

Installed packages to be REMOVED:
	tailscale: 1.0.3

Number of packages to be removed: 1

The operation will free 17 MiB.
[1/1] Deinstalling tailscale-1.0.3...
[1/1] Deleting files for tailscale-1.0.3: 100%
**root@OPNsense**:/usr/ports/security/tailscale # make reinstall
===>  Deinstalling for tailscale
===>   tailscale not installed, skipping
===>  Installing for tailscale-1.2.0
===>  Checking if tailscale is already installed
===>   Registering installation for tailscale-1.2.0
Installing tailscale-1.2.0...
**root@OPNsense**:/usr/ports/security/tailscale # service tailscaled enable
tailscaled enabled in /etc/rc.conf
**root@OPNsense**:/usr/ports/security/tailscale # service tailscaled start
**root@OPNsense**:/usr/ports/security/tailscale # tailscale version
date.20200921
  go version: go1.15.4

**root@OPNsense**:/usr/ports/security/tailscale # cd
**root@OPNsense**:~ # pkg info tailscale-1.2.0
tailscale-1.2.0
Name           : tailscale
Version        : 1.2.0
Installed on   : Fri Nov 13 14:35:51 2020 CET
Origin         : security/tailscale
Architecture   : FreeBSD:12:amd64
Prefix         : /usr/local
Categories     : security
Licenses       : BSD3CLAUSE
Maintainer     : mikael@FreeBSD.org
WWW            : https://tailscale.com/
Comment        : Mesh VPN that makes it easy to connect your devices
Options        :
        PIE            : on
        RELRO          : on
Annotations    :
        FreeBSD_version: 1201000
Flat size      : 16.9MiB
Description    :
Tailscale is a mesh VPN alternative, based on WireGuard, that connects your
computers, databases, and services together securely without any proxies.

WWW: https://tailscale.com/


opnsense = OPNsense 20.7.4-amd64 | FreeBSD 12.1-RELEASE-p10-HBSD | OpenSSL 1.1.1h 22 Sep 2020 | on old sophos SG210
tailscale = 1.2.0 installed like i described it before

**Comment by bramevo** [comment]
Since Wireguard is supported by OPNsense, I would imagine that using Tailscale should be possible. Maybe the clue can be found in the OPNsense Wireguard section.

**Comment by bramevo** [comment]
Just did some tests. Installing using the method of @newmy-de worked perfectly till the activation of tailscale.
It installed tailscale-1.2.0 using the make install, so I did not have to upgrade it after installation.

What is failing after the installation:

    tailscale up --advertise-tags tag:something --authkey tskey-......

Seems to hang forever on that command without any response.
Same thing for

    tailscale up

Tailscale netcheck response:

    tailscale netcheck    
    Report:
	* UDP: true
	* IPv4: yes, 78.23........:60385
	* IPv6: no
	* MappingVariesByDestIP: false
	* HairPinning: false
	* PortMapping: not checked
	* Nearest DERP: Frankfurt
	* DERP latency:
		- fra: 20.7ms  (Frankfurt)
		- lhr: 25.1ms  (London)
		- nyc: 103.7ms (New York City)
		- blr: 149.4ms (Bangalore)
		- sfo: 161.3ms (San Francisco)
		- dfw: 168.3ms (Dallas)
		- sea: 173.8ms (Seattle)
		- tok: 229.1ms (Tokyo)
		- sin: 273ms   (Singapore)
		- syd: 313.4ms (Sydney)

In OPNSense logs:

    2020-12-16T17:22:38 | kernel | tun0: changing name to 'tailscale0' |  
    -- | -- | -- | --
    2020-12-16T17:22:38 | kernel | tun0: link state changed to UP

Might be that the firewall stops it (although I'm not sure why because rules should allow all traffic). Are there any specific IPs from tailscale that I need to look for?

**Comment by kivikakk** [comment]
I had a similar issue trying Tailscale v1.2.0 on FreeBSD (12.2-RELEASE), where `tailscale up` hangs forever. Building v1.2.10 from source and using that worked great 👍

**Comment by Bramzor** [comment]
> I had a similar issue trying Tailscale v1.2.0 on FreeBSD (12.2-RELEASE), where `tailscale up` hangs forever. Building v1.2.10 from source and using that worked great 👍

Latest version of OPNSense is running on FreeBSD 12.1-RELEASE-p11-HBSD #0 74f1f081a1e(stable/20.7)-dirt .

**Comment by DentonGentry** [comment]
Confirming that I had the same problem with the up command hanging on OPNsense with 1.2.0, and that updating to 1.2.10 resolved it. In my case, in /usr/ports/security/tailscale I have:

```
root@OPNsense:/usr/ports/security/tailscale # git diff Makefile distinfo
diff --git a/security/tailscale/Makefile b/security/tailscale/Makefile
index 2fd8a5b53c6..ae339e789d3 100644
--- a/security/tailscale/Makefile
+++ b/security/tailscale/Makefile
@@ -1,7 +1,7 @@
 # $FreeBSD$

 PORTNAME=      tailscale
-PORTVERSION=   1.2.0
+PORTVERSION=   1.2.10
 DISTVERSIONPREFIX=     v
 CATEGORIES=    security

@@ -30,7 +30,7 @@ GH_TUPLE=     \
                pborman:getopt:ee0cd42419d3:pborman_getopt/vendor/github.com/pborman/getopt \
                peterbourgon:ff:v2.0.0:peterbourgon_ff_v2/vendor/github.com/peterbourgon/ff/v2 \
                rsc:goversion:v1.2.0:rsc_goversion/vendor/rsc.io/goversion \
-               tailscale:wireguard-go:a6168fd06b3f:tailscale_wireguard_go/vendor/github.com/tailscale/wireguard-go \
+               tailscale:wireguard-go:db78fad0bebf:tailscale_wireguard_go/vendor/github.com/tailscale/wireguard-go \
                tcnksm:go-httpstat:v0.2.0:tcnksm_go_httpstat/vendor/github.com/tcnksm/go-httpstat \
                toqueteos:webbrowser:v1.2.0:toqueteos_webbrowser/vendor/github.com/toqueteos/webbrowser

diff --git a/security/tailscale/distinfo b/security/tailscale/distinfo
index bb34f0a6f49..8c997fe9e6f 100644
--- a/security/tailscale/distinfo
+++ b/security/tailscale/distinfo
@@ -1,6 +1,6 @@
 TIMESTAMP = 1604134461
-SHA256 (tailscale-tailscale-v1.2.0_GH0.tar.gz) = 7af324d3d1cdae6b71e3ecbbbc3240359b053530e388d539b4a45405c8ceeb9e
-SIZE (tailscale-tailscale-v1.2.0_GH0.tar.gz) = 436605
+SHA256 (tailscale-tailscale-v1.2.10_GH0.tar.gz) = 3d06b68fe588fe623aa6a4df1cc10fffdbaf58d0fe4b92bdb81a6ab879616fc8
+SIZE (tailscale-tailscale-v1.2.10_GH0.tar.gz) = 441791
 SHA256 (apenwarr-fixconsole-5a9f6489cc29_GH0.tar.gz) = 60a7c20f3c7253049c70bcef2d5cfbc4f33eea5250e7f820eb48748b0a505418
 SIZE (apenwarr-fixconsole-5a9f6489cc29_GH0.tar.gz) = 6349
 SHA256 (go4org-mem-185c595c3ecc_GH0.tar.gz) = 7c2e249ff20c395664f8dd9fdf7842ba1d6b444d7d0c1751b4aaf3b847867079
@@ -31,8 +31,8 @@ SHA256 (peterbourgon-ff-v2.0.0_GH0.tar.gz) = f281421980071b90b5525900d4471fcf94f
 SIZE (peterbourgon-ff-v2.0.0_GH0.tar.gz) = 25295
 SHA256 (rsc-goversion-v1.2.0_GH0.tar.gz) = 65e9c181995ed8c9207ea08d79a148ab53307b7fad2172a5a0764d6aa393a98e
 SIZE (rsc-goversion-v1.2.0_GH0.tar.gz) = 8586
-SHA256 (tailscale-wireguard-go-a6168fd06b3f_GH0.tar.gz) = d38540d713e1265454c48959c786c46da8cee506cd77430c2578db260e054d90
-SIZE (tailscale-wireguard-go-a6168fd06b3f_GH0.tar.gz) = 115401
+SHA256 (tailscale-wireguard-go-db78fad0bebf_GH0.tar.gz) = 5abc21da43eafb6ef52667af522dc62547eab20e640ffdb64ebfbb8f89323349
+SIZE (tailscale-wireguard-go-db78fad0bebf_GH0.tar.gz) = 103091
 SHA256 (tcnksm-go-httpstat-v0.2.0_GH0.tar.gz) = 42f6fe0985afde00d0317b9be4b6c9bf22df300ce596a3b47547597a5f44662b
 SIZE (tcnksm-go-httpstat-v0.2.0_GH0.tar.gz) = 5957
 SHA256 (toqueteos-webbrowser-v1.2.0_GH0.tar.gz) = e48ffbcbc8490d4f393a70f82d4c579c4dcde5d5bbf1553ff159522a2df2bdd4
root@OPNsense:/usr/ports/security/tailscale #
```

I did this manually, updating the expected size and SHA256. Previous updates [in FreshPorts](https://www.freshports.org/security/tailscale), like [this one](https://svnweb.freebsd.org/ports/head/security/tailscale/distinfo?r1=544614&r2=553757), appear to update all of the go modules and probably have an automated method to do it. If I can figure out how to do that, I'll followup on freshports.

**Comment by DentonGentry** [comment]
Filed https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=252429 asking to update the port to 1.2.10.

**Comment by DentonGentry** [comment]
FreeBSD accepted the diffs to update Tailscale from 1.2.0.to 1.2.10.
https://svnweb.freebsd.org/changeset/ports/560496 (via https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=252429)

If I understand it correctly it will need to go into Hardened BSD in their next regular update from ports, and then into OPNsense from Hardened BSD.

**Comment by DentonGentry** [comment]
The update to tailscale 1.2.10 made it into the OPNsense ports on 1/6/2021:

```
commit 0265e226fe74f95f2a0e046fb84a17b0e2df831d
Author: Franco Fichtner <franco@opnsense.org>
Date:   Wed Jan 6 21:02:51 2021 +0100

    */*: sync with upstream

    Taken from: HardenedBSD
```

**Comment by bradfitz** [comment]
Closing, as this seems done. There are even docs now: https://tailscale.com/kb/1097/install-opnsense

**Comment by patmaddox** [comment]
This seems to have regressed in 1.6.0 on FreeBSD. I did manage to install 1.2.10 from source, but the 1.6.0 build has the same infinite hang problem.

**Comment by DentonGentry** [comment]
A comment was left about version 1.6, since deleted, so for future reference: we don't know of any issues with Tailscale 1.6 or later on FreeBSD. The FreeBSD /usr/ports/security/tailscale was updated to use Tailscale 1.6 on March 25, 2021. I've been running it on OPNsense (HardenedBSD 12.1) for some time.

If you do run into trouble with `tailscale up` just hanging, a couple things to try:
1. ktrace what it is doing
1. `mv /var/db/tailscale/tailscaled.state /var/db/tailscale/tailscaled.state.prev` and `service restart tailscaled` to have it make a new one. If this fixes anything please post to this issue, we'll want to know how that happened.

**Comment by DentonGentry** [comment]
Filed https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=255829 to update HardenedBSD Tailscale port from 1.6.0 to 1.8.3. Tested on OPNsense 21.1.5.

**Comment by DickensCider** [comment]
Love your product, would be nice to be able to use it ;)

I think its great that the tailscale team are working with certain products/developers to help integrate tailscale. Having said that, purely a guess, but I would think that there are way more opensense/pfsense users that fit the tailscale demographic than say synology users. It would be great to see a package come out for both op/pf in the near future so that we all don't have to figure out op/pf work arounds that may, or may not, cause issues for other things we have set up in our networks

Cheers,
Dickens
pfsense user

**Comment by mrnoisytiger** [comment]
Is there any chance there could be updated instructions for PfSense? These OPNSense instructions do not seem applicable to PfSense.

**Comment by DentonGentry** [comment]
Some progress: https://github.com/DentonGentry/FreeBSD-ports/commits/tailscale

It doesn't work yet:
- Authentication page renders but doesn't work.
- State apparently isn't saved in a way the pfSense init system likes, tailscaled isn't being started after reboot.

**Comment by DentonGentry** [comment]
Thanks to https://twitter.com/chrisleppanen/status/1433493827533246465, the state is now being saved in a way which the pfSense init system likes.

There is still more to be done about getting the system authenticated.

**Comment by mrnoisytiger** [comment]
@DentonGentry Really appreciate the work on the package! 

Might be jumping the gun a little bit, but would it be possible to use the pfSense instance as a subnet router as well? Thanks!

**Comment by DentonGentry** [comment]
FreeBSD has supported operation as a subnet router for a long time. In Tailscale 1.14 we added support for operating as an exit node (https://github.com/tailscale/tailscale/issues/2498). I expect pfSense to work as a subnet router and as an exit node.

**Comment by KaiDevrim** [comment]
How would one go about installing TailScale in pfSense?

**Comment by DentonGentry** [comment]
pfSense as a platform is not finished and not supported by Tailscale. If you want to try to work on it, the pfSense ports tree we've been working from is https://github.com/DentonGentry/FreeBSD-ports/

**Comment by srk** [comment]
Any chance of distributing binaries for OpnSense, rather than just doing a local build? The official procedure works fine for me until 'make  install', caused by golang.org not resolving for me for location-dependent reasons. Even when I fetch the souce code tarball by other means and drop it in /usr/ports/distfiles, all the dependencies are resolved as URLs with host proxy.golang.org in them, so they fail. I tried starting bash and setting the GOPROXY, HTTP_PROXY and HTTPS_PROXY env variables to a known country-specific mirror, but that didn't work either, go the package couldn't even build. Any practical solutions to this?

**Comment by foxalerts23** [comment]
Interested in tailscale for pfSense.

**Comment by RT4U** [comment]
> Interested in tailscale for pfSense.

Same would love to have it.

**Comment by NZPy7h0n** [comment]
Been a few months, any updates?

**Comment by DentonGentry** [comment]
I promise that we will update the bug when work begins in earnest. This is not currently at the top of the list to work on.

If anyone reading this thread has experience working on pfSense extensions and takes contract work: we'd write a contract to implement a Tailscale pfSense extension.

**Comment by zeroZshadow** [comment]
From what I understand, the package already exists, but needs to be evaluated for inclusion into the package system by pfSense.
https://github.com/pfsense/FreeBSD-ports/tree/devel/security/tailscale

**Comment by DentonGentry** [comment]
That is the port from FreeBSD. pfSense does not use the FreeBSD init system, it has its own init system constructed in PHP. Tailscale doesn't start at boot using the FreeBSD port alone. Additionally, support for Tailscale in the pfSense UI would need additional work.

There is a start on that additional pfSense support in https://github.com/DentonGentry/FreeBSD-ports/tree/tailscale, but it doesn't really work well. If anyone reading this thread has experience working on pfSense extensions and takes contract work: we'd pay a contract to implement a Tailscale pfSense extension.

**Comment by kode54** [comment]
Possibly related, I followed the above directions to get it installed on OPNsense, but only the default routes are working. It isn't configuring accepted routes from the Tailscale network I'm joined to.

**Comment by DentonGentry** [comment]
`tailscale up --accept-routes` ?

**Comment by kode54** [comment]
Routes are being accepted. I had to upgrade my OPNsense to 22.7 beta with the 13.1 kernel for the other routes to be added properly.

Now I have all the routes up, but I can't ping any of the other hosts in my network. Ever since I swapped out the network interfaces of the host machine from vmxnet3 to virtio. Is there something I need to do to tailscale to make it refresh its network configuration?

**Comment by kode54** [comment]
I can ping 100.100.100.100 from both the router, and from my LAN. But I can't reach any of my network's nodes from anywhere on any of my devices. Not even from a local device if I start Tailscale client on it.

**Comment by jeryd** [comment]
> From what I understand, the package already exists, but needs to be evaluated for inclusion into the package system by pfSense. https://github.com/pfsense/FreeBSD-ports/tree/devel/security/tailscale

Any update on the timeline for this?

**Comment by DentonGentry** [comment]
As stated earlier in this bug, I promise that we will update it when work begins in earnest.

We posted on Twitter offering a contract for someone familiar with pfSense plugin development to help.

**Comment by mayakacz** [comment]
Netgate has shared they are working on a pfsense package: https://www.youtube.com/watch?v=Fg_jIPVcioY

**Comment by theonemcdonald** [comment]
Hi all,

Package will be available in the package manager for both pfSense Plus and Community Edition soon. Code will be available in our CE ports tree (https://github.com/pfsense/freebsd-ports).

Best,
Christian

**Comment by gonzopancho** [comment]
> As stated earlier in this bug, I promise that we will update it when work begins in earnest.
> 
> We posted on Twitter offering a contract for someone familiar with pfSense plugin development to help.

You did, and then your company was largely unresponsive, even when we published a screenshot of the work that Christian had (obviously) already begun.

Anyway, the first version should drop soon.

**Comment by forkwhilefork** [comment]
I love it when Jim Thompson is rude on the internet. We should make a subreddit for jimspotting. Bonus points if it's directed at a company that previously wanted to work with his company.

**Comment by theonemcdonald** [comment]
Package has dropped

Have a great weekend all

**Comment by zeroZshadow** [comment]
Gave it a test, and it works nicely for me, although it initially wouldn't authenticate.
I did have to manually set it to not use the tailscale dns settings through the command line. So if others want to use this, note that tailscale accepts DNS settings by default.

Thanks for the awesome work @theonemcdonald It's immensely appreciated!

**Comment by SGC1990** [comment]
I am still trying to get nat to route traffic out over a dedicated ip. Other than that works great. @theonemcdonald

**Comment by SGC1990** [comment]
> Gave it a test, and it works nicely for me, although it initially wouldn't authenticate. I did have to manually set it to not use the tailscale dns settings through the command line. So if others want to use this, note that tailscale accepts DNS settings by default.
> 
> Thanks for the awesome work @theonemcdonald It's immensely appreciated!

What do you mean you had to set tailscale dns setting thought command line.

**Comment by SGC1990** [comment]
> Gave it a test, and it works nicely for me, although it initially wouldn't authenticate. I did have to manually set it to not use the tailscale dns settings through the command line. So if others want to use this, note that tailscale accepts DNS settings by default.
> 
> Thanks for the awesome work @theonemcdonald It's immensely appreciated!

What do you mean you had to set tailscale dns setting

**Comment by zeroZshadow** [comment]
> > Gave it a test, and it works nicely for me, although it initially wouldn't authenticate. I did have to manually set it to not use the tailscale dns settings through the command line. So if others want to use this, note that tailscale accepts DNS settings by default.
> > Thanks for the awesome work @theonemcdonald It's immensely appreciated!
> 
> What do you mean you had to set tailscale dns setting thought command line.

Since there is no option for this yet, I had to invoke `tailscale up .... --accept-dns=false` to ensure its not overwriting the dns settings with 100.100.100.100 from tailscale, which is not something I wanted for my setup.

**Comment by theonemcdonald** [comment]
That's trivial to implement. Will fix.

**Comment by n8henrie** [comment]
Following this issue for months, installed without a hitch on 2 boxes within minutes of the availability announcement. Thank you!

**Comment by frezbo** [comment]
is there plans for a similar plugin for opnsense rather than installing from a port? I'm willing to sponsor up to 100$

**Comment by DentonGentry** [comment]
@frezbo we'll track that in https://github.com/tailscale/tailscale/issues/5067

**Comment by DentonGentry** [comment]
I believe this feature request is done, thank you @[theonemcdonald](https://github.com/theonemcdonald) and @[gonzopancho](https://github.com/gonzopancho)

**Comment by jeryd** [comment]
Do I need to set up any rules for the firewall / tailscale / pfsense?

**Comment by theonemcdonald** [comment]
For traffic destined for the tailscale interface address, yes. You can create rules in the Tailscale interface group for that. Cross site traffic is invisible to pf because of how subnet router works on FreeBSD. This appears to be locally sourced by the time it's on wire.

**Comment by jeryd** [comment]
> For traffic destined for the tailscale interface address, yes. You can create rules in the Tailscale interface group for that. Cross site traffic is invisible to pf because of how subnet router works on FreeBSD. This appears to be locally sourced by the time it's on wire.

Got it. My use case is to use tailscale to access pf when I am not at home. Thanks.

**Comment by bradfitz** [comment]
@theonemcdonald, I'm happy to help you get FreeBSD kernel-level subnet router/exit node support working. If you have any questions about the codebase or whatnot, send me an email (my github username `@tailscale.com`) or https://calendly.com/bradfitz and we can do a quick video call whenever.

**Comment by brunoga** [comment]
So I have Tailscale up on OPNSense and it works as long as I am logged in to the router itself (then I can access all my Tailscale nodes from it). It would be great if I could route packets from my local network computers (connected to that router) to the Tailscale network so I could contact other nodes with them (without having them also being Tailscale nodes) but, as I understand it, this is not possible right now with OPNSense (or FreeBSD more generally). Is supporting this in the roadmap by any chance?

**Comment by DentonGentry** [comment]
Could you file that as a new issue? We don't keep track of closed bugs.
I would expect `tailscale up --advertise-routes=...` to work. If it doesn't we'll need to look into it.
